web/modules/custom/simplytest_projects/src/ProjectFetcher.php

```diff
@@ -79,6 +82,14 @@ public function __construct(
    * @todo should not return null, but throw exceptions.
    */
   public function fetchProject(string $shortname): ?array {
+    // Sanitize shortname for use in lock key: allow only lowercase letters, numbers, and underscores.
+    $sanitized_shortname = preg_replace('/[^a-z0-9_]/', '', $shortname);
+    if (!$this->lock->acquire("fetch_project_$sanitized_shortname")) {
```

Comment:
Lock name needs to be unique otherwise it just gets extended

Reply:
Technically it is unique per request with `\Drupal\Core\Lock\LockBackendAbstract::getLockId` just not in tests.